haskell/cabal
# `default-package-bounds` (RFC #9569) implementation

Solves #8912 #9569 

Docs: https://cabal--9570.org.readthedocs.build/en/9570/cabal-package-description-file.html#default-package-bounds

# Manual QA

By adding a field `default-package-bounds` in a cabal file and a list of dependencies with version ranges, for each of those dependencies, any dependency declaration that was unbounded will be set to the one on `default-package-bounds`:

```
default-package-bounds
  build-depends:
    text > 5

library a 
  build-depends: text

library b 
  build-depends: text >5.5

library c
  build-depends: y
```

Should be understood as:

```
library a 
  build-depends: text >5

library b 
  build-depends: text >5.5

library c
  build-depends: y
```

If the stanza is omitted, then nothing changes wrt previous version.

Notice this should not influence:
- `pkgconfig-depends`
- transitive dependencies
- setup-depends

--------------------------------------------------
